gapipy/models/base.py

```diff
@@ -164,21 +155,19 @@ def _set_resource_field(self, field, value):
             setattr(self, field, self._model_cls(field)(value, client=self._client, stub=True))
 
     def _set_resource_collection_field(self, field, value):
-        is_parent_resource = getattr(self, '_is_parent_resource', None)
-        if is_parent_resource:
+        # set the `parent` attribute
+        parent = None
```

Comment:
Why would `_is_parent_resource` be missing?

Reply:
🤔 I didn't really look beyond simplifying the existing code, but you bring up a good point. Technically it should always be present. The `_resource_collection_field` property is used in the case where we have an `href` pointing to a paginated list of sub/nested resources.

e.g. Bookings are a parent resource and primarily link to nested resources such as `services`, `requirements`, etc.

It could be there are cases where a `BaseModel` object may link to a `nested` list, and in that case the `BaseModel` would not have a `parent` attribute, so then the conundrum of how would be build the `uri`. I'm not seeing any immediate code gapipy that would break if we remove this check, and always assume there is a `parent`, but I'm not 💯 certain.

Reply:
Interesting! Initially I was going to suggest an EAFP solution if it really had a reason to be missing, but then I wondered whether the assumption that it could be missing was valid— since handling it creates the implication that it's okay or expected